ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveSortJoinReduceRule.java

```diff
@@ -111,10 +113,17 @@ public void onMatch(RelOptRuleCall call) {
     RelNode inputLeft = join.getLeft();
     RelNode inputRight = join.getRight();
 
+    final RexBuilder rexBuilder = sortLimit.getCluster().getRexBuilder();
+    // We have to retain 0 ~ offset + limit because each task might not access the global offset
```

Comment:
What does `0 ~ offset + limit` means?
Is it `Each task has to retain offset + limit records because ...` ?

Reply:
You're right. I removed the comment since it a little duplicated with the newly added comment